### PR TITLE
Ratpack micrometer #1389

### PR DESCRIPTION
**Pull request by mustaine**
~WIP

I have started ratpack-micrometer using prometheus. #1389
I based some abstraction on ratpack-dropwizard.

Let's discuss implementation details from here?

---
This change is [Reviewable](https://reviewable.io/reviews/ratpack/ratpack/1404)

### CODE REVIEW

**Comment by drmaas** [comment]
@johnrengelman fyi

**Comment by beckje01** [comment]
This is really cool, I'm wondering if it makes sense to release this in its own repo first get a few integrations working then bring it into the main repo? Doesn't seem to depend on changes in core etc

File: ratpack-core/src/main/java/ratpack/handling/HandlerDecorator.java [hunk 1]
```diff
@@ -163,7 +163,7 @@ static HandlerDecorator noop() {
    * As the {@code rest} argument encapsulates the application handlers, the returned handler should generally delegate to it (via {@link Context#insert(Handler...)}).
    *
    * @param serverRegistry the server registry
-   * @param rest the rest of the handlers of the application
+   * @param rest           the rest of the handlers of the application
```
Comment (drmaas) [inline ratpack-core/src/main/java/ratpack/handling/HandlerDecorator.java]: formatting?

**Comment by johnrengelman** [comment]
@mustaine can you rebase this PR to the latest master?

**Comment by jkwatson** [comment]
Seems like it would be better to implement via opencensus, rather than a specific metric consumer. https://opencensus.io/

**Comment by drmaas** [comment]
@jkwatson that was my main point - micrometer itself is capable of reporting dimensional metrics to many different backends (telegraf and influxdb for example), not just prometheus.

**Comment by jkwatson** [comment]
@drmaas yep. Just wanted to point out that opencensus seems to be heading toward being a pretty good standard library to use.  👍

**Comment by beckje01** [comment]
This was merged in #1439

**Pull request closed without merging**